spec/rspec/mocks/stub_spec.rb

```diff
@@ -350,6 +363,21 @@ class << self; public :hello; end;
           expect(mod.hello).to eq(:hello)
         end
 
+        it "correctly restores from allow_any_instance_of for self extend" do
```

Comment:
This test can catch the need for `!Mocks.space.any_instance_recorder_for(owner, true)` in the other condition. It was discovered in https://gitlab.com/gitlab-org/gitlab/-/merge_requests/44417#note_429658296 thanks to @engwan !